api/api.go

```diff
@@ -6,40 +6,153 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"strconv"
 
+	"github.com/ONSdigital/dp-authorisation/auth"
+	dphandlers "github.com/ONSdigital/dp-net/handlers"
+	dprequest "github.com/ONSdigital/dp-net/request"
 	"github.com/ONSdigital/dp-topic-api/apierrors"
 	"github.com/ONSdigital/dp-topic-api/config"
+	"github.com/ONSdigital/dp-topic-api/store"
 	"github.com/ONSdigital/log.go/log"
 	"github.com/gorilla/mux"
 )
 
+var (
+	trueStringified = strconv.FormatBool(true)
+
+	createPermission = auth.Permissions{Create: true}
+	readPermission   = auth.Permissions{Read: true}
+	updatePermission = auth.Permissions{Update: true}
+	deletePermission = auth.Permissions{Delete: true}
+)
+
 //API provides a struct to wrap the api around
 type API struct {
-	Router  *mux.Router
-	mongoDB MongoServer
+	Router                 *mux.Router
+	dataStore              store.DataStore
+	topicPermissions       AuthHandler
+	permissions            AuthHandler
+	enablePrivateEndpoints bool
 }
 
 //Setup function sets up the api and returns an api
-func Setup(ctx context.Context, cfg *config.Config, r *mux.Router, mongoDB MongoServer) *API {
+func Setup(ctx context.Context, cfg *config.Configuration, router *mux.Router, dataStore store.DataStore, topicPermissions AuthHandler, permissions AuthHandler) *API {
 	api := &API{
-		Router:  r,
-		mongoDB: mongoDB,
+		Router:                 router,
+		dataStore:              dataStore,
+		topicPermissions:       topicPermissions,
+		permissions:            permissions,
+		enablePrivateEndpoints: cfg.EnablePrivateEndpoints,
 	}
 
-	//!!! see dp-image for possible best code ...
 	if cfg.EnablePrivateEndpoints {
 		// create publishing related endpoints ...
-		//!!! adjust this for authorisation, and have comment that states 'add authorisation', etc.
-		r.HandleFunc("/topics/{id}", api.GetTopicHandler).Methods(http.MethodGet)
+		log.Event(ctx, "enabling private endpoints for topic api", log.INFO)
+
+		api.enablePrivateTopicEndpoints(ctx)
 	} else {
 		// create web related endpoints ...
-		r.HandleFunc("/topics/{id}", api.GetTopicHandler).Methods(http.MethodGet)
+
+		log.Event(ctx, "enabling only public endpoints for dataset api", log.INFO)
+		api.enablePublicEndpoints(ctx)
 	}
 
-	r.HandleFunc("/hello", HelloHandler()).Methods("GET")
+	router.HandleFunc("/hello", HelloHandler()).Methods("GET")
```

Comment:
Maybe the `/hello` endpoint can now be removed? (it's a placeholder generated by the project generation tool)

Reply:
yes, i'll remove that in a separate PR